File: frontend/src/types/notification.ts
export interface Notification {
    id: number;
    title: string;
    message: string;
    type: NotificationType;
    related_id?: number;
    read: boolean;
    read_at?: string;
    created_at: string;
}

export type NotificationType = 
    | "session_scheduled"
    | "session_reminder"
    | "session_cancelled"
    | "match_suggested"
    | "match_accepted"
    | "feedback_received"
    | "system";

export interface NotificationSettings {
    email_notifications: boolean;
    push_notifications: boolean;
    session_reminders: boolean;
    match_notifications: boolean;
    feedback_notifications: boolean;
    reminder_time: number; // minutos antes de la sesión
}

export interface NotificationFilters {
    unread_only?: boolean;
    type?: NotificationType;
    startDate?: Date;
    endDate?: Date;
}

export interface NotificationStats {
    total: number;
    unread: number;
    today: number;
}

File: backend/app/schemas/notification.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class NotificationBase(BaseModel):
    title: str
    message: str
    type: str
    related_id: Optional[int] = None

class NotificationCreate(NotificationBase):
    user_id: int

class NotificationResponse(NotificationBase):
    id: int
    user_id: int
    read: bool
    read_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True

class NotificationSettings(BaseModel):
    email_notifications: bool = True
    push_notifications: bool = True
    session_reminders: bool = True
    match_notifications: bool = True
    feedback_notifications: bool = True
    reminder_time: int = 30  # minutos antes de la sesión
